sockeye/translate.py

```diff
@@ -110,16 +131,14 @@ def translate_lines(output_handler: sockeye.output_handler.OutputHandler, source
 
     i = 0
     total_time = 0.0
-    for i, line in enumerate(source_data, 1):
-        tic = time.time()
-        trans_input = translator.make_input(i, line)
-        logger.debug(" IN: %s", trans_input)
-        trans_output = translator.translate(trans_input)
-        trans_wall_time = time.time() - tic
-        total_time += trans_wall_time
-        logger.debug("OUT: %s", trans_output)
-        logger.debug("OUT: time=%.2f", trans_wall_time)
-        output_handler.handle(trans_input, trans_output, trans_wall_time)
+    tic = time.time()
+    trans_inputs = []
+    for i, line in enumerate(source_data, 1):  # TODO not using comprehesions to keep the last i; fix this
```

Comment:
why not use a list comprehension? you can just get i from the length of trans_inputs, which is a constant operation.

Reply:
Good point, was an artefact of having inputs previously as generators, not lists

Reply:
Why not use generators? (just curious what this entailed)

Reply:
We need one pass over inputs to see if some sentences need splitting before we send them for translation in batches. At this point it's possible to do it with generators, at the cost of slightly more complicated code, but we could have it a separate PR. 

The real mess will start if we decide to fill batches optimally. Then, some of the sentence chunks might end up in different batches. The code will have to keep track of them for later gluing, not being able to output anything over potentially a very long time, storing intermediate results in memory until some of them can be output. In the worst case this may require storing everything until the last batch completes.

Reply:
Ok, this makes a lot of sense :)
Lets stay with lists then.